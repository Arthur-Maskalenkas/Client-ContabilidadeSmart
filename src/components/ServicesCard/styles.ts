import styled, { css } from 'styled-components'

export const Wrapper = styled.article`
  ${({ theme }) => css`
    display: flex;
    flex-direction: column;
    padding: ${theme.spacings.large};
    box-shadow: 0px 0px 50px -2px rgba(0, 0, 0, 0.52);
  `}
`

export const Content = styled.div`
  ${({ theme }) => css`
    margin-top: ${theme.spacings.small};
  `}
`

export const Title = styled.h2`
  ${({ theme }) => css`
    color: ${theme.colors.black}
    line-height: 3.4rem;
    font-size: ${theme.font.sizes.small};
    font-weight: ${theme.font.family.poppins.weight.bold};
  `}
`

export const Description = styled.p`
  ${({ theme }) => css`
    color: ${theme.colors.gray};
    line-height: 2.7rem;
    font-size: 1.5rem;
    font-weight: ${theme.font.family.poppins.weight.normal};
  `}
`

export const SaibaMais = styled.p`
  ${({ theme }) => css`
    color: ${theme.colors.black};
    font-size: 1.5rem;
    font-weight: ${theme.font.family.poppins.weight.bold};
  `}
`
