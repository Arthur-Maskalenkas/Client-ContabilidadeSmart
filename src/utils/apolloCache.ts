import { InMemoryCache } from '@apollo/client'
import { concatPagination } from '@apollo/client/utilities'

// A query QueryGames retorna -> games <-, e é isso que o cache vai acumular games

// ** fluxo:
// > Eu pego uma lista de free (retorna games assim como varias outras query)
// > o cache fica [game1,game2,game3]
// > Assim que outra parte usa alguma query que retorne game, ela vai receber independente de tudo [game1,game2,game3], e não [game4,game5,game6]
// **

// Para quebrar o fluxo, eu separei o cacheamento por 'where' e 'sort'. Assim fica algo como: (caso puxe 3 games de 3 queries diferentes)

//**
// > games: {"where: {"release_date_lte":"2021-01-29"}, "sort":"release_date:desc"}: [] 8 items"}
// > games: {"where: {"release_date_gt":"2021-01-29"}, "sort":"release_date:asc"}: [] 8 items"}
// > games: {"where: {"price": 0}}: [] 8 items
// > Antes ficaria tudo em games com 24 items
//**
export default new InMemoryCache({
  typePolicies: {
    Query: {
      fields: {
        posts: concatPagination(['where', 'sort'])
      }
    }
  }
})
