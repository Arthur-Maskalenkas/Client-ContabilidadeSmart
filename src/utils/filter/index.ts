import { ParsedUrlQueryInput } from 'querystring'

type ParseArgs = {
  queryString: ParsedUrlQueryInput
}

export const parseQueryStringToWhere = ({ queryString }: ParseArgs) => {
  // eslint-disable-next-line @typescript-eslint/no-explicit-any
  const obj: any = {}

  // queryString = {platforms: ['windows','linux']}
  // filterItems = {name: 'platforms', type: 'checkbox'}
  Object.keys(queryString).forEach((key) => {
    // Pegando o valor e colocando dentro do objeto
    obj[key] = { slug_contains: queryString[key] }
  })

  return obj
}
