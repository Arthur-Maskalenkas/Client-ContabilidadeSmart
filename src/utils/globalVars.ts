export const IMAGE_NOT_FOUND =
  'https://res.cloudinary.com/dezwlfeyb/image/upload/v1635853776/102-022_800x_ebxk1t.jpg'

export const LINK_SITE = 'localhost:3000'

// --  API

export const API_LINKEDIN =
  'https://www.linkedin.com/shareArticle?mini=true&url='

export const API_FACEBOOK = 'https://www.facebook.com/sharer/sharer.php?u='

export const API_WHATSAPP = 'https://api.whatsapp.com/send?text='

// --  API
