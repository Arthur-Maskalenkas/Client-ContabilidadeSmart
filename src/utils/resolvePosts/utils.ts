export const operations = {
  nextPage: nextPageOperation(),
  backPage: backPageOperation()
}

type variablesType = {
  limit?: number
  start?: number
}

export type FetchMoreProps = {
  variables: variablesType
}
