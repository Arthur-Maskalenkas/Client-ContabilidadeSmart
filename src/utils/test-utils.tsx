import { ReactElement } from 'react'
import { render, RenderOptions } from '@testing-library/react'

import { ThemeProvider } from 'styled-components'
import theme from 'styles/theme'

type CustomRenderProps = Omit<RenderOptions, 'queries'>

const customRender = (ui: ReactElement, { ...renderOptions }: CustomRenderProps = {}) =>
  render(<ThemeProvider theme={theme}>{ui}</ThemeProvider>, renderOptions)

// Tudo que eu chamar, vai vim dessa pasta aqui
export * from '@testing-library/react'
export { customRender as render }
