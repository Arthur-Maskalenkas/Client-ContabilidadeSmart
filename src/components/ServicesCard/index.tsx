import Link from 'next/link'

import IconsWrapper, { IconTypes } from 'components/IconsWrapper'
import * as S from './styles'

import { resolveUrl } from 'utils/resolveUrl'

export type ServicesCardProps = {
  icon: IconTypes
  title: string
  description: string
  slug: string
}

const ServicesCard = ({
  icon,
  description,
  title,
  slug
}: ServicesCardProps) => (
  <S.Wrapper>
    <IconsWrapper icon={icon} size="6rem" />

    <S.Content>
      <S.Title>{title}</S.Title>
      <S.Description>{description}</S.Description>

      <Link passHref href={resolveUrl('artigos', slug)}>
        <S.SaibaMais>Saiba mais</S.SaibaMais>
      </Link>
    </S.Content>
  </S.Wrapper>
)

export default ServicesCard
