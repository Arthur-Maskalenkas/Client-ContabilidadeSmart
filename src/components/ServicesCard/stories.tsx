import { Story, Meta } from '@storybook/react'
import ServicesCard, { ServicesCardProps } from '.'

import { MINIMAL_VIEWPORTS } from '@storybook/addon-viewport'
import { Container } from 'components/Container'

export default {
  title: 'ServicesCard',
  component: ServicesCard
} as Meta

export const Default: Story<ServicesCardProps> = (args) => (
  <Container>
    <div
      style={{
        marginTop: '50px',
        display: 'grid',
        gridTemplateColumns: '1fr  1fr 1fr',
        gap: '5rem'
      }}
    >
      <ServicesCard {...args} icon="Facebook" />
      <ServicesCard {...args} icon="Instagram" />
      <ServicesCard {...args} icon="Whatsapp" />
    </div>
  </Container>
)

Default.args = {
  title: 'Infoproduto',
  description:
    'Youtubers, Produtores, Coprodutores, Especialistas / Experts, Afiliados, Gerente de Afiliados, Eugência, Agência de Lançamentos, Gestor de Tráfego, Copywriter, Designer Gráfico e Video Maker são as Profissões do Futuro.',
  icon: 'Whatsapp'
}

Default.parameters = {
  backgrounds: {
    default: 'mm-light'
  }
}

export const Mobile: Story<ServicesCardProps> = (args) => (
  <div
    style={{
      maxWidth: '90vw',
      marginLeft: 'auto',
      marginRight: 'auto',
      marginTop: '50px'
    }}
  >
    <ServicesCard {...args} />
  </div>
)

Mobile.args = {
  title: 'Infoproduto',
  description:
    'Youtubers, Produtores, Coprodutores, Especialistas / Experts, Afiliados, Gerente de Afiliados, Eugência, Agência de Lançamentos, Gestor de Tráfego, Copywriter, Designer Gráfico e Video Maker são as Profissões do Futuro.',
  icon: 'Whatsapp'
}

Mobile.parameters = {
  backgrounds: {
    default: 'mm-light'
  },
  viewport: {
    viewports: MINIMAL_VIEWPORTS,
    defaultViewport: 'mobile1'
  }
}
