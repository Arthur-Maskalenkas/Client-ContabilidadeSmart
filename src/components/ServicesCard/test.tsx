import { render, screen } from 'utils/test-utils'

import ServicesCard, { ServicesCardProps } from '.'

const props: ServicesCardProps = {
  title: 'Um titulo',
  description: 'Uma descriçao',
  slug: 'Um slug',
  icon: 'Facebook'
}

describe('<ServicesCard />', () => {
  it('should render o componente', () => {
    const { container } = render(<ServicesCard {...props} />)

    expect(
      screen.getByRole('heading', { name: /um titulo/i })
    ).toBeInTheDocument()

    expect(screen.getByText(/Uma descriçao/i)).toBeInTheDocument()

    expect(
      screen.getByRole('link', {
        name: /saiba mais/i
      })
    ).toBeInTheDocument()

    expect(container).toMatchSnapshot()
  })
})
