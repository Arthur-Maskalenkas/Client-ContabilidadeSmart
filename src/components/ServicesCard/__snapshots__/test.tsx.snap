// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ServicesCard /> should render o componente 1`] = `
.c4 {
  display: inline-block;
  vertical-align: middle;
  overflow: hidden;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-transition: all 0.5s;
  transition: all 0.5s;
}

.c2 > .c3 {
  width: 6rem;
  color: #FFFFFF;
  position: relative;
}

.c0 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  min-height: 50rem;
  overflow: hidden;
  cursor: pointer;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  padding: 4.0rem;
  box-shadow: 0px 0px 24px -16px #000000;
}

.c0 .c1 {
  padding: 1.5rem;
  border-radius: 50%;
  width: 8rem;
  background-color: #F66A3E;
}

.c0 .c3 {
  color: white;
}

.c0:hover .c6,
.c0:hover .c9 {
  color: #FFFFFF;
}

.c0:hover .c1 {
  -webkit-transform: scale(1.2);
  -ms-transform: scale(1.2);
  transform: scale(1.2);
}

.c0:after {
  content: '';
  width: 100%;
  height: 100%;
  background: #000;
  position: absolute;
  top: 0;
  opacity: 0;
  visibility: hidden;
  left: 0;
  z-index: -1;
  -webkit-transition: 0.1;
  transition: 0.1;
  -webkit-transform: scale(0.9);
  -ms-transform: scale(0.9);
  transform: scale(0.9);
}

.c0:hover:after {
  opacity: 0.75;
  visibility: visible;
  -webkit-transform: scale(1);
  -ms-transform: scale(1);
  transform: scale(1);
}

.c0:before {
  content: '';
  width: 100%;
  height: 100%;
  background: rgb(0 0 0 / 0.5);
  position: absolute;
  top: 0;
  left: -100%;
  opacity: 1;
  z-index: -1;
  -webkit-transition: 300ms;
  transition: 300ms;
}

.c0:hover:before {
  left: 0;
  -webkit-transition: 300ms;
  transition: 300ms;
  -webkit-transition-delay: 200ms;
  transition-delay: 200ms;
}

.c5 {
  margin-top: 2.4rem;
}

.c7 {
  color: #050706 line-height:3.4rem;
  font-size: 2rem;
  font-weight: 600;
}

.c8 {
  color: #8F8F8F;
  line-height: 2.7rem;
  font-size: 1.5rem;
  font-weight: 400;
}

.c10 {
  color: #050706;
  font-size: 1.5rem;
  font-weight: 600;
}

<div>
  <article
    class="c0"
  >
    <div
      class="c1 c2"
      color="white"
    >
      <svg
        aria-hidden="true"
        class="c3 c4"
        fill="currentColor"
        focusable="false"
        role="svg"
        viewBox="0 0 24 24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M13.397 20.997v-8.196h2.765l.411-3.209h-3.176V7.548c0-.926.258-1.56 1.587-1.56h1.684V3.127A22.336 22.336 0 0 0 14.201 3c-2.444 0-4.122 1.492-4.122 4.231v2.355H7.332v3.209h2.753v8.202h3.312z"
        />
      </svg>
    </div>
    <div
      class="c5"
    >
      <h2
        class="c6 c7"
      >
        Um titulo
      </h2>
      <p
        class="c8"
      >
        Uma descri√ßao
      </p>
      <a
        class="c9 c10"
        href="/artigo/Um%20slug"
      >
        Saiba mais
      </a>
    </div>
  </article>
</div>
`;
